src/BuildQueue.cpp

```diff
@@ -911,6 +921,25 @@ namespace t2
         result = ExecuteProcess(cmd_line, env_count, env_vars, thread_state->m_Queue->m_Config.m_Heap, job_id, false, SlowCallback, &slowCallbackData);
         passedOutputValidation = ValidateExecResultAgainstAllowedOutput(&result, node_data);
       }
+
+      bool bad = false;
+
+      for (int i = 0; i < outputFiles.GetCount(); i++)
+      {
+        FileInfo info = GetFileInfo(outputFiles[i].m_Filename);
+
+        if (pre_timestamps[i] == info.m_Timestamp)
+        {
+          Log(kError, "Error: the %d'th output (which was %s) of %s did not change. Every action must touch all its outputs.",
```

Comment:
I think we should find a way to get this failure information into the ExecuteProcessResult,  so that it clearly shows on a failed build together with the stdout, stderr, exitcode etc.